pkg/hintrunner/hinter/scope.go

```diff
@@ -84,17 +105,25 @@ func (sm *ScopeManager) GetVariableValueAsBigInt(name string) (*big.Int, error)
 		return nil, err
 	}
 
-	valueBig, ok := value.(*big.Int)
-	if !ok {
-		return nil, fmt.Errorf("value: %s is not a *big.Int", value)
-	}
+	// valueBig, ok := value.(*big.Int)
+	// if !ok {
+	// 	return nil, fmt.Errorf("value: %s is not a *big.Int", value)
+	// }
 
-	return valueBig, nil
+	return value.BigIntValue, nil
 }
 
-func (sm *ScopeManager) getCurrentScope() (*map[string]any, error) {
+func (sm *ScopeManager) getCurrentScope() (*ScopeMap, error) {
 	if len(sm.scopes) == 0 {
 		return nil, fmt.Errorf("expected at least one existing scope")
 	}
 	return &sm.scopes[len(sm.scopes)-1], nil
 }
+
+func BigIntScopeValue(value *big.Int) *ScopeValue {
+	return &ScopeValue{Type: TypeBigInt, BigIntValue: value}
```

Comment:
To minimize allocations on the heap, try not to return pointers to objects allocated locally in function whenever possible.
So in this case it would be better to return a value instead of pointer

Reply:
Done